### PR TITLE
Fix creating-a-product link in content views page

### PR DESCRIPTION
**Pull request by ianballou**
The link currently give a 404.  Needed to updated content.html to index.html.

### CODE REVIEW

**Comment by jturel** [comment]
Looks like 3.14 is missing so this branch probably needs a rebase!

File: plugins/katello/2.4/user_guide/content_views/index.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ What can a Content View be used for?
 
 ## General Workflow
 
-First [create a product and repository](/plugins/katello/{{ page.version }}/user_guide/content/content.html#creating-a-product) in the library environment and populate the repository with content (by syncing it or uploading content).
+First [create a product and repository](/plugins/katello/{{ page.version }}/user_guide/content/index.html#creating-a-product) in the library environment and populate the repository with content (by syncing it or uploading content).
```
Comment (jturel) [inline plugins/katello/2.4/user_guide/content_views/index.md:28]: This view also has a 'Lifecycle Environments' link that's also broken. Can you fix it too? It has the identical problem of not referring to index.html

Comment (ianballou) [reply plugins/katello/2.4/user_guide/content_views/index.md:28]: sure thing!

Comment (ianballou) [reply plugins/katello/2.4/user_guide/content_views/index.md:28]: I found more places where broken links were used, I replaced them all.

**Review by jturel — CHANGES_REQUESTED** [review]
I can't believe we had so many broken links.

I found another: in puppet_integration/index.html the 'content views guide' link is broken. Mind fixing this one too? Any others we find we can fix later - this PR is big enough :)

**Comment by ianballou** [comment]
@jturel fixed it!

**Review by jturel — APPROVED** [review]
ACK thanks @ianballou !

**Pull request merged**